homeassistant/components/google_assistant/__init__.py

```diff
@@ -93,7 +94,16 @@ def _check_report_state(data):
 async def async_setup(hass: HomeAssistant, yaml_config: Dict[str, Any]):
     """Activate Google Actions component."""
     config = yaml_config.get(DOMAIN, {})
-    google_config = async_register_http(hass, config)
+
+    google_store = GoogleConfigStore(hass)
```

Comment:
Let's move this into a new `GoogleConfig.async_initialize` method. No reason to make consumers of the config know about this.

Reply:
It makes testing painful. you need to patch all the time to avoid the store, and currently the testing uses global fixed default config structure (BASIC_CONFIG of MockConfig) that use a None hass object. So with store calling on the hass object it's messy.

Ps. Yes i could theoretically switch that BASIC_CONFIG to a fixture. Which would simplify things. But that would grow this pull more than i like.

Reply:
I don't understand, you could just patch `BASIC_CONFIG` to appear loaded ?

Reply:
Will give it a try, had problems with that before. Code has changed since then (the GoogleConfigStore didn't exist). So it might work now.

Reply:
You can also just load the store on demand inside the add_agent_user_id function? No reason to have it load before it's being used ?

Reply:
I didn't even hit that problem this time. Moved creation of the Store object into async_initialize() that way i avoided the dependency in the __init__ functions.